res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,452 @@
+                    if (control === 52 && loopSize >= (1/16)) { //turn left
+                        engine.setValue(this.group, "beatloop_size", loopSize / 2);
+                    } else if (control === 53 && loopSize < 256) { // turn right
+                        engine.setValue(this.group, "beatloop_size", loopSize * 2);
+                    }
+                };
+            } else {
+                this.input = function(channel, control, _value, _status, _group) { // if set to true invert function
+                    var direction = (control === 52) ? "Down" : "Up"; //turn left: turn right
+                    script.triggerControl("[Library]", "Move" + direction);
+                };
+            }
+        },
+        shift: function() {
+            if (!NumarkScratch.invertLoopEncoderFunction) { //if set to false then normal function
+                this.input = function(channel, control, _value, _status, _group) {
+                    var direction = (control === 52) ? "Down" : "Up"; //turn left: turn right
+                    script.triggerControl("[Library]", "Move" + direction);
+                };
+            } else {
+                this.input = function(channel, control, _value, _status, _group) { // if set to true invert function
+                    var loopSize = engine.getValue(this.group, "beatloop_size");
+                    if (control === 52 && loopSize >= (1/16)) { //turn left
+                        engine.setValue(this.group, "beatloop_size", loopSize / 2);
+                    } else if (control === 53 && loopSize < 256) { //turn  right
+                        engine.setValue(this.group, "beatloop_size", loopSize * 2);
+                    }
+                };
+            }
+        },
+    });
+
+    this.loopEncoderButton = new components.Button({
+        shift: function() {
+            if (!NumarkScratch.invertLoopEncoderFunction) { //if set to false then normal function
+                this.inKey = "LoadSelectedTrack";
+            } else {
+                this.inKey = "beatloop_activate"; // if set to true invert function
+            }
+        },
+        unshift: function() {
+            if (!NumarkScratch.invertLoopEncoderFunction) { //if set to false then normal function
+                this.inKey = "beatloop_activate";
+            } else {
+                this.inKey = "LoadSelectedTrack"; // if set to true invert function
+            }
+        }
+    });
+
+    this.padSection = new NumarkScratch.PadSection(number);
+
+    this.shiftButton = new components.Button({
+        input: function(channel, control, value) {
+            if (this.isPress(channel, control, value)) {
+                NumarkScratch.shift();
+                midi.sendShortMsg(0x9F, 0x32, 0x7F);
+            } else {
+                NumarkScratch.unshift();
+                midi.sendShortMsg(0x9F, 0x32, NumarkScratch.LOW_LIGHT);
+            }
+        }
```

Comment:
More idiomatic solution:
```suggestion
        input: function(channel, control, value) {
            midi: [0x9F, 0x32],
            if (this.isPress(channel, control, value)) {
                NumarkScratch.shift();
                this.send(this.on);
            } else {
                NumarkScratch.unshift();
                this.send(this.off);
            }
        }
```